blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx

```diff
@@ -80,12 +80,34 @@ export class RunDetailsPipeline extends Component {
     }
 
     componentWillReceiveProps(nextProps) {
-        if (this.props.result.isQueued()) {
+        if (nextProps.result.isQueued()) {
             return;
         }
         const followAlong = this.state.followAlong;
         this.mergedConfig = this.generateConfig({ ...nextProps, followAlong });
-
+        
+        if (this.mergedConfig.supportsNode && (!nextProps.nodes || !nextProps.nodes[this.mergedConfig.nodeKey])) {
+            const tryNodes = () => {
+                // in order to handle queued items during re-run, we need to retry fetching nodes periodically
+                if (!this.props.nodes || !this.props.nodes[this.mergedConfig.nodeKey]) {
```

Comment:
Guessing `this.props` is right ... maybe a comment wrt the difference/relationship between `this.props` and `nextProps` and what's being waited for here. Waiting for a node (karaoke?) associated with the rerun to appear or something?